dev/src/reference.ts

```diff
@@ -1365,10 +1472,44 @@ export class Query<T = firestore.DocumentData> implements firestore.Query<T> {
     fieldPath: string | firestore.FieldPath,
     opStr: firestore.WhereFilterOp,
     value: unknown
+  ): Query<T>;
+
+  /**
+   * Creates and returns a new [Query]{@link Query} with the additional filter
+   * that documents should satisfy the relation constraint(s) provided.
+   *
+   * This function returns a new (immutable) instance of the Query (rather than
+   * modify the existing instance) to impose the filter.
+   *
+   * @param {Filter} filter A unary or composite filter to apply to the Query.
+   * @returns {Query} The created Query.
+   *
+   * @example
+   * ```
+   * let collectionRef = firestore.collection('col');
+   *
+   * collectionRef.where(Filter.and(Filter.where('foo', '==', 'bar'), Filter.where('foo', '!=', 'baz'))).get()
+   *   .then(querySnapshot => {
+   *     querySnapshot.forEach(documentSnapshot => {
+   *       console.log(`Found document at ${documentSnapshot.ref.path}`);
+   *     });
+   * });
+   * ```
+   */
+  where(filter: Filter): Query<T>;
+
+  where(
+    fieldPathOrFilter: string | firestore.FieldPath | Filter,
+    opStr?: firestore.WhereFilterOp,
+    value?: unknown
   ): Query<T> {
```

Comment:
Just a note that this is like the `.filter` function in datastore

Reply:
I'm not sure on the history, but I agree the naming conventions deviate between Firestore and Datastore.